Post

mitogh opened the pull request:
- **What kind of change does this PR introduce?** Feature
- **What is the current behavior?** None at the moment
- **What is the new behavior (if this is a feature change)?** Explained in readme.md
- **Does this PR introduce a breaking change?** Nop.
- **Other information**:

@zzuga @adamf321 Please take a look and let me know if you have any comments.

Thanks.

mitogh committed:
Update travis config and composer file

mitogh committed:
Update the base class

mitogh committed:
Create a class that creates a collection endpoint

mitogh committed:
Helper class to Sanitize values

mitogh committed:
Fix typo

mitogh committed:
Prevent to deliver insecure data to the endpoint

mitogh committed:
Use default callbacks from WP

mitogh committed:
Fix CI issues

mitogh committed:
Reme Sanitize collection to prevent collisions

mitogh committed:
Add pagination information on the ednpoint

mitogh committed:
Update instructions

mitogh committed:
Avoid tracking changes on Mac file

mitogh committed:
Update list of examples

mitogh committed:
Add filter used on this class

mitogh committed:
Allow to format the data send to the endpoint

mitogh committed:
Update instructions and add examples

mitogh committed:
Update default data

zzuga commented on src/Collection/Sanitize.php:
```diff
@@ -0,0 +1,23 @@
+		$map = function( $item ) {
+			return sanitize_text_field( $item );
+		};
+		return array_map( $map, $data );
```
Do we need a closure here? Doesnt this work?
  return array_map( 'sanitize_text_field', $data );

mitogh replied:
Thanks for the tip, and technically `sanitize_text_field` is a named clousre isn`t?

> On Mar 31, 2016, at 5:07 AM, Caridad Zapatero Contreras notifications@github.com wrote:
> 
> In src/Collection/Sanitize.php https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58036339:
> 
> > - \* @package Lean\Endpoints
> > - */
> >   +class Sanitize {
> >   +
> > -   /**
> > -    \* Function that iterates over an array to Sanitize the value of the array
> > -    \* in secure text values.
> > -    *
> > -    \* @param array $data The data to be Sanitized.
> > -    \* @return array The cleaned data.
> > -    */
> > -   public static function text_array( array $data = [] ) {
> > -       $map = function( $item ) {
> > -           return sanitize_text_field( $item );
> > -       };
> > -       return array_map( $map, $data );
> >   Do we need a closure here? Doesnt this work?
> >   return array_map( 'sanitize_text_field', $data );
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/86356df199c5bac0bd0cfbda93ff685a610a4205#r58036339

zzuga commented on composer.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "moxie-leean/wp-endpoints-archive",
+  "name": "moxie-lean/wp-endpoints-archive",
   "description": "wp-endpoints-archive",
```
Why the different naming (Collection vs Archive)?

mitogh replied:
Basically the main thing with the endpoints is to make it agnostic from WP Since are going to be used from a FE that usually does not know nothing about WP, archive makes sense in terms of WP outside of a WP world not su much, but if you have any other suggestions we can discuss about the naming.

Thanks.

> On Mar 31, 2016, at 5:19 AM, Caridad Zapatero Contreras notifications@github.com wrote:
> 
> In composer.json https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58037571:
> 
> > @@ -1,5 +1,5 @@
> >  {
> > -  "name": "moxie-leean/wp-endpoints-archive",
> > -  "name": "moxie-lean/wp-endpoints-archive",
> >   "description": "wp-endpoints-archive",
> >   Why the different naming (Collection vs Archive)?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/86356df199c5bac0bd0cfbda93ff685a610a4205#r58037571

adamf321 replied:
Shouldn't composer.json say collection instead of archive?

mitogh replied:
Correct my bad.

> On Mar 31, 2016, at 6:58 AM, Adam Fenton notifications@github.com wrote:
> 
> In composer.json https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58048496:
> 
> > @@ -1,5 +1,5 @@
> >  {
> > -  "name": "moxie-leean/wp-endpoints-archive",
> > -  "name": "moxie-lean/wp-endpoints-archive",
> >   "description": "wp-endpoints-archive",
> >   Shouldn't composer.json say collection instead of archive?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/86356df199c5bac0bd0cfbda93ff685a610a4205#r58048496

mitogh committed:
Remove not required class anymore

adamf321 commented on readme.md:
````diff
@@ -0,0 +1,165 @@
+The easiest way to install this package is by using composer from your terminal:
+
+```bash
+composer require moxie-lean/wp-endpoints-archive
````
Why are you using the name archive still?

mitogh replied:
Same as before.

> On Mar 31, 2016, at 6:59 AM, Adam Fenton notifications@github.com wrote:
> 
> In readme.md https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58048633:
> 
> > @@ -0,0 +1,165 @@
> > +# WP Endpoint Collection
> > +
> > +> Query against a WordPress site and get a JSON response with a
> > +> collection of data associated with your request.
> > +
> > +## Getting Started
> > +
> > +The easiest way to install this package is by using composer from your terminal:
> > +
> > +```bash
> > +composer require moxie-lean/wp-endpoints-archive
> > Why are you using the name archive still?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/79212b057c9e0b6231ca448bc9f4e34930f2b65e#r58048633

adamf321 commented on readme.md:
````diff
@@ -0,0 +1,165 @@
+/wp-json/leean/v1/collection
+```
+
+By default the collection is the list of all the posts, you can use most
````
Should we have a default posts_per_page, eg 10? Otherwise you might accidentally pull out 1000's of records.

mitogh replied:
By default uses the number of post of reading options.

> On Mar 31, 2016, at 7:00 AM, Adam Fenton notifications@github.com wrote:
> 
> In readme.md https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58048760:
> 
> > -  state that is not publish.
> >   +- Avoid to query post with passwords protections.
> >   +
> >   +By default returns only a collection of posts but any post type can be
> >   +specifed to be returned or a collection of multiple post types can be
> >   +specifed as well.
> >   +
> >   +## Usage.
> >   +
> >   +The default URL is:
> >   +
> >   +``
> >   +/wp-json/leean/v1/collection
> >   +`
> >   +
> >   +By default the collection is the list of all the posts, you can use most
> >   Should we have a default posts_per_page, eg 10? Otherwise you might accidentally pull out 1000's of records.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/79212b057c9e0b6231ca448bc9f4e34930f2b65e#r58048760

adamf321 replied:
That's fine. Can you update the readme? It says that it will return **all** posts.

mitogh replied:
Yes, is specified here: 

https://github.com/moxie-leean/wp-endpoints-collection/blob/post/readme.md#usage

adamf321 replied:
I meant this line, it should say "By default the collection is the list of <posts per page> posts, "

mitogh replied:
Ok thanks, I'm going to update the doc.

adamf321 commented on readme.md:
````diff
@@ -0,0 +1,165 @@
+
+```json
+wp-json/leean/v1/collection?post_type[]=post&post_type[]=page
+```
````
What about "page"? We need a way of specifying which page we're querying (1 by default)

mitogh replied:
I’m not sure what you mean here. Would you mind giving more details.

> On Mar 31, 2016, at 7:01 AM, Adam Fenton notifications@github.com wrote:
> 
> In readme.md https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58048917:
> 
> > +
> > +`json
> > +wp-json/leean/v1/collection?author=1
> > +`
> > +
> > +Get the all the posts ordered by ID from lowest to highest values.
> > +
> > +`json
> > +wp-json/leean/v1/collection?orderby=ID&order=ASC
> > +`
> > +
> > +Get all the posts and pages published on the site.
> > +
> > +`json
> > +wp-json/leean/v1/collection?post_type[]=post&post_type[]=page
> > +`
> > What about "page"? We need a way of specifying which page we're querying (1 by default)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/79212b057c9e0b6231ca448bc9f4e34930f2b65e#r58048917

adamf321 replied:
I mean paged (as it's called in WP_Query) - the page number you want to query -1, 2, 3, etc. We should add an example with this.

mitogh replied:
Example added.

adamf321 commented on readme.md:
```diff
@@ -0,0 +1,165 @@
+- `$the_post`, this is a `WP_Post` object that allow you to easily
+  access to the ID, title, content or any other data of the current
+item.
+- `$args`, this are the arguments used to create the request, for
```
These should be included by default

mitogh replied:
Don’t think so, should be the other way instead.

> On Mar 31, 2016, at 7:11 AM, Adam Fenton notifications@github.com wrote:
> 
> In readme.md https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58050207:
> 
> > +add_filter( 'ln_collection_enable_filter_format', '__return_true' );
> > +```
> > +
> > +And if you want to disable the format you only need to remove the
> > +filter.
> > +
> > +`ln_collection_item_format`. This filter is executed only if
> > +`ln_collection_enable_filter_format`has been updated as mentioned
> > +before. This filter has different params.
> > +
> > +-`$default_format`, this is an array with the default information that
> > -  every endpoint returns
> >   +- `$the_post`, this is a `WP_Post` object that allow you to easily
> > -  access to the ID, title, content or any other data of the current
> >   +item.
> >   +- `$args`, this are the arguments used to create the request, for
> >   These should be included by default
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/79212b057c9e0b6231ca448bc9f4e34930f2b65e#r58050207

adamf321 replied:
I think it should to help with debugging. To me this is more important than saving a few bytes.
And also consistent with the post endpoint (which returns the slug)

mitogh replied:
Ok. What part exactly to be clear?

Thanks.

adamf321 replied:
The input $_REQUEST params

mitogh replied:
The args are included by default, when we add a function to that filter we can just add the params that we are going to use, args is always passed but the user can avoid using it .

adamf321 replied:
ok, got it!

adamf321 commented on src/Collection/Filter.php:
```diff
@@ -0,0 +1,12 @@
+ * @since 0.1.0
+ */
+class Filter {
+	const ITEM_IS_ENABLED = 'ln_collection_enable_filter_format';
```
We should prefix with ln_endpoints_ to be consitent

mitogh replied:
The constant name?

> On Mar 31, 2016, at 7:12 AM, Adam Fenton notifications@github.com wrote:
> 
> In src/Collection/Filter.php https://github.com/moxie-leean/wp-endpoints-collection/pull/2#discussion_r58050346:
> 
> > @@ -0,0 +1,12 @@
> > +<?php namespace Lean\Endpoints\Collection;
> > +
> > +/**
> > - \* Custom filters used on this endpoint.
> > - *
> > - \* @package Endpoints\Collection;
> > - \* @since 0.1.0
> > - */
> >   +class Filter {
> > -   const ITEM_IS_ENABLED = 'ln_collection_enable_filter_format';
> >   We should prefix with ln_endpoints_ to be consitent
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub https://github.com/moxie-leean/wp-endpoints-collection/pull/2/files/79212b057c9e0b6231ca448bc9f4e34930f2b65e#r58050346

adamf321 replied:
no, the filter name - eg ln_endpoints_collection_enable_filter_format

adamf321 commented on src/Collection.php:
```diff
@@ -0,0 +1,158 @@
+
+use Leean\AbstractEndpoint;
+
+use Lean\Endpoints\Collection\Sanitize;
```
where is this?

mitogh committed:
Update module name

mitogh committed:
Update filter name

mitogh committed:
Update docs by adding more examples

mitogh committed:
Pass the request arguments to the filter

In order to be more dynamic and make tests agains the queries that
create the endpoint collection

mitogh committed:
Remove not required line from travis

mitogh commented:
Let me know if there is anything else we need to address here.

mitogh committed:
Update instructions

mitogh commented:
Updated!
